keps/sig-node/1967-size-memory-backed-volumes/README.md

```diff
@@ -0,0 +1,263 @@
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+This KEP improves the portability of pod definitions that use memory backed empty dir
+volumes by sizing an empty dir memory backed volume as the minimum of pod allocatable
+memory on a host and an optional explicit user provided value.
+
+## Motivation
+
+Kubernetes supports emptyDir volumes whose backing storage is memory (i.e. tmpfs).
+The size of this memory backed volume is defaulted to 50% of the memory on a Linux host.
+The coupling of default memory backed volume size with the host that runs the pod makes
+pod definitions less portable across node instance types and providers.
+
+This impacts workloads that make heavy use of /dev/shm or other use cases oriented around
+memory backed volume usage (AI/ML, etc.)
+
+### Goals
+
+- Size a memory backed volume to match the pod allocatable memory
+- Enable a user to size the memory backed volume less than the pod allocatable memory
+
+### Non-Goals
+
+- Address memory chargeback of empty dir volumes across container restarts
+
+## Proposal
+
+Define a new feature gate: `SizeMemoryBackedVolumes`.
+
+If enabled, the `kubelet` will change the behavior when building memory backed
+volume to specify a non-zero size that is the following:
+
+`min(nodeAllocatable[memory], podAllocatable[memory], emptyDir.sizeLimit)`
+
+This is an improvement over present behavior as pods will see emptyDir memory
+backed volumes sized based on actual allowed usage rather than a heuristic
+based on the node that is executing the pod.
+
+### Risks and Mitigations
+
+The risks for this proposal are minimal.
+
+The empty dir volume will now be sized consistently with pod level cgroup
+memory limit.  A container that writes to a memory backed volume is charged
+for that write while accounting memory.  If a container restarts, the charge
+goes to the pod cgroup.  Sizing the emptyDir volume to match the actual amount
+of memory that can be charged to a pod basically avoids undersizing or oversizing
+the appearance of more memory.
+
+## Design Details
+
+The design for this implementation makes the existing `emptyDir.sizeLimit`
+not just used during eviction heuristics, but for sizing of the volume.
```

Comment:
To be precise, general memory or disk eviction don't take emptyDir.sizeLimit into account.  But as part of local ephemeral storage capacity isolation, we evict pods where the usage of an empty-dir exceeds its sizeLimit